x-pack/plugins/ui_actions_enhanced/public/drilldowns/url_drilldown/components/url_drilldown_collect_config/url_drilldown_collect_config.tsx

```diff
@@ -0,0 +1,193 @@
+    } catch {
+      return urlTemplate;
+    }
+  }, [urlTemplate, scope]);
+  const scopeVariables = React.useMemo(() => buildScopeSuggestions(scope), [scope]);
+
+  function updateUrlTemplate(newUrlTemplate: string) {
+    if (config.url.template !== newUrlTemplate) {
+      onConfig({
+        ...config,
+        url: {
+          ...config.url,
+          template: newUrlTemplate,
+        },
+      });
+    }
+  }
+  const { error, isValid } = React.useMemo(
+    () => validateUrlTemplate({ template: urlTemplate }, scope),
+    [urlTemplate, scope]
+  );
+  const isEmpty = !urlTemplate;
+  const isInvalid = !isValid && !isEmpty;
+  return (
+    <>
+      <EuiFormRow
+        fullWidth
+        isInvalid={isInvalid}
+        error={error}
+        className={'uaeUrlDrilldownCollectConfig__urlTemplateFormRow'}
+        label={txtUrlTemplateLabel}
+        helpText={
+          syntaxHelpDocsLink && (
+            <EuiLink external target={'_blank'} href={syntaxHelpDocsLink}>
+              {txtUrlTemplateHelpLinkText}
+            </EuiLink>
+          )
+        }
+        labelAppend={
+          <AddVariableButton
+            variables={scopeVariables}
+            onSelect={(variable: string) => {
+              if (textAreaRef.current) {
+                updateUrlTemplate(
+                  urlTemplate.substr(0, textAreaRef.current!.selectionStart) +
+                    `{{${variable}}}` +
+                    urlTemplate.substr(textAreaRef.current!.selectionEnd)
+                );
+              } else {
+                updateUrlTemplate(urlTemplate + `{{${variable}}}`);
+              }
+            }}
+          />
+        }
+      >
+        <EuiTextArea
+          fullWidth
+          isInvalid={isInvalid}
+          name="url"
+          data-test-subj="urlInput"
+          value={urlTemplate}
+          placeholder={txtUrlTemplatePlaceholder}
+          onChange={(event) => updateUrlTemplate(event.target.value)}
+          rows={3}
+          inputRef={textAreaRef}
+        />
+      </EuiFormRow>
+      <EuiFormRow
+        fullWidth
+        label={txtUrlTemplatePreviewLabel}
+        labelAppend={
+          <EuiText size="xs">
+            <EuiLink href={compiledUrl} target="_blank" external>
+              {txtUrlTemplatePreviewLinkText}
+            </EuiLink>
+          </EuiText>
+        }
+      >
+        <EuiTextArea
+          fullWidth
+          name="urlPreview"
+          data-test-subj="urlPreview"
+          value={compiledUrl}
+          disabled={true}
+          rows={3}
+        />
+      </EuiFormRow>
+      <EuiFormRow hasChildLabel={false}>
+        <EuiCheckbox
+          id="openInNewTab"
+          name="openInNewTab"
+          label={txtUrlTemplateOpenInNewTab}
+          checked={config.openInNewTab}
+          onChange={() => onConfig({ ...config, openInNewTab: !config.openInNewTab })}
+        />
+      </EuiFormRow>
+    </>
+  );
+};
+
+function AddVariableButton({
```

Comment:
Shall we type this React component as React component and put it in is its own file?

```suggestion
AddVariableButton: React.FC<...>
```

Reply:
I wouldn't move it into separate file as don't consider it a separate enough piece to justify extracting it. 
(Probably matter of preference? For me it is easier to navigate code in a single file, then jumping around multiple files and in this case. So I tend to not prematurely create new files)